integration/src/integration/java/pl/allegro/tech/hermes/integration/JmsConsumingTest.java

```diff
@@ -64,6 +65,43 @@ public void shouldPublishAndConsumeJMSMessageWithTraceId() throws Exception {
         assertThat(jmsEndpoint.waitAndGetLastMessage()).assertStringProperty("TraceId", traceId);
     }
 
+    @Test
+    public void shouldPublishAndConsumeJmsMessageWithTraceHeaders() throws Exception {
+
+        // given
+        String message = "{\"hello\": \"world\"}";
+        String traceId = UUID.randomUUID().toString();
+        String spanId = UUID.randomUUID().toString();
+        String parentSpanId = UUID.randomUUID().toString();
+        String sampled = "1";
+        String reported = "0";
+
+        // and
+        Topic topic = operations.buildTopic("publishJmsGroupWithTrace", "topic");
+        operations.createSubscription(topic, "subscription", jmsEndpointAddress(JMS_TOPIC_NAME));
+        WebTarget client = ClientBuilder.newClient().target(FRONTEND_URL).path("topics").path(topic.getQualifiedName());
+        jmsEndpoint.expectMessages(TestMessage.of("hello", "world"));
+
+        // when
+        Response response = client
+                .request()
+                .header("Trace-Id", traceId)
+                .header("Span-Id", spanId)
+                .header("Parent-Span-Id", parentSpanId)
+                .header("Trace-Sampled", sampled)
+                .header("Trace-Reported", reported)
+                .post(Entity.entity(message, MediaType.APPLICATION_JSON));
+
+        // then
+        assertThat(response).hasStatus(Response.Status.CREATED);
+        Message lastMessage = jmsEndpoint.waitAndGetLastMessage();
+        assertThat(lastMessage).assertStringProperty("TraceId", traceId);
+        assertThat(lastMessage).assertStringProperty("SpanId", spanId);
+        assertThat(lastMessage).assertStringProperty("ParentSpanId", parentSpanId);
+        assertThat(lastMessage).assertStringProperty("TraceSampled", sampled);
+        assertThat(lastMessage).assertStringProperty("TraceReported", reported);
```

Comment:
That's an actually very good question, since according to the specs: https://docs.oracle.com/javaee/6/api/javax/jms/Message.html - the underscore is valid indentifier also see http://www-01.ibm.com/support/knowledgecenter/SSDKJ8_6.2.0/com.ibm.wpg.entadv.doc/config/integration/creating_jms_properties_bo.html

But due to the existing convention of other header handling I've implemented this in the same manner.